### PR TITLE
Add windows CI

### PR DESCRIPTION
**Pull request by Illviljan**
I get a lot of errors locally on windows.

### CODE REVIEW

**Comment by dcherian** [comment]
OK This is going to be a lot easier on a windows machine. What is the default dtype for `np.array([1, 2, 3])`

**Comment by Illviljan** [comment]
```python
import numpy as np

a = np.array([1, 2, 3])

a
Out[43]: array([1, 2, 3])

a.dtype
Out[44]: dtype('int32')
```

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -43,11 +43,14 @@ jobs:
...
-          python -m pip install -e .
-          conda list
+          python -m pip install --no-deps -e .
       - name: Run Tests
         run: |
           pytest -n auto --cov=./ --cov-report=xml
+      - name: Version info
+        run: |
+          conda info -a
+          conda list
```
Comment (dcherian) [inline .github/workflows/ci.yaml:53]: This should already be in the setup-micromamba bit.
```suggestion
```

File: flox/core.py [hunk 2]
```diff
@@ -1784,4 +1784,9 @@ def groupby_reduce(
 
     if _is_minmax_reduction(func) and is_bool_array:
         result = result.astype(bool)
+
+    # we lose dtype information when creating a pandas Index
```
Comment (dcherian) [inline flox/core.py:1788]: Nice, this was a real bug it seems like.

Comment (dcherian) [reply flox/core.py:1788]: Actually this is a bad idea. The core issue is that for `cohorts` and `blockwise` we use `np.unique` and concatenate, so dtype is preserved. For `map-reduce` we convert to a pandas Index so it always casts. (this is when expected_groups is None).

I think the solution is to always cast to whatever type pandas likes until we can specify dtypes for pd.Index

**Review by Illviljan — COMMENTED** [review]
LGTM, nice @dcherian ! :)

**Pull request merged**